File: Node/Prisma_ORM/src/bll/services/supplierService.js
const SupplierRepository = require('../../dal/repositories/supplierRepository');

const SupplierService = {
  getAll: async () => {
    return await SupplierRepository.getAll();
  },

  getById: async (id) => {
    const entity = await SupplierRepository.getById(id);
    if (!entity) throw new Error('entity not found');
    return entity;
  },

  create: async (entity) => {
    return await SupplierRepository.create(entity);
  },

  update: async (id, entity) => {
    return await SupplierRepository.update(id, entity);
  },

  delete: async (id) => {
    return await SupplierRepository.delete(id);
  }
};

module.exports = SupplierService; 

File: Go/GORM_ORM/bll/services/supplierService.go
package services

import (
	"GORM_ORM/dal/entities"
	"GORM_ORM/dal/repositories"
)

type SupplierService interface {
	GetAll() ([]entities.Supplier, error)
	GetByID(id int) (*entities.Supplier, error)
	Create(supplier *entities.Supplier) error
	Update(supplier *entities.Supplier) error
	Delete(id int) error
}

type supplierService struct {
	repo repositories.SupplierRepository
}

func NewSupplierService(repo repositories.SupplierRepository) SupplierService {
	return &supplierService{repo: repo}
}

func (s *supplierService) GetAll() ([]entities.Supplier, error) {
	return s.repo.GetAll()
}

func (s *supplierService) GetByID(id int) (*entities.Supplier, error) {
	return s.repo.GetByID(id)
}

func (s *supplierService) Create(supplier *entities.Supplier) error {
	return s.repo.Create(supplier)
}

func (s *supplierService) Update(supplier *entities.Supplier) error {
	return s.repo.Update(supplier)
}

func (s *supplierService) Delete(id int) error {
	return s.repo.Delete(id)
}
